spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinitionToDslConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringEscapeUtils;
+
+/**
+ * Reverse engineers a {@link TaskDefinition} into a semantically equivalent DSL text representation.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class TaskDefinitionToDslConverter {
```

Comment:
Given how tricky the reverse engineering could be we must have a proper Test class for this converter. 
Similar to StreamDefinitionToDslConverterTests we should validate the behavior with different expressions, characters or regexes.